luminoso_api/client.py

```diff
@@ -395,13 +379,13 @@ def change_path(self, path):
             url = self.root_url + path
         else:
             url = self.url + path
-        return self.__class__(self._auth, url)
+        return self.__class__(self.session.auth, url)
```

Comment:
All this making of new Session objects.  Would it be better to have the constructor take a session?